internal/cmd/headless_repl/src/processes/process_completion.rs

```diff
@@ -0,0 +1,115 @@
+use crate::processes::process_completion::HintType::{
+    ArgumentType, FunctionType, MethodType, PackageType, UnimplementedType,
+};
+use crate::CommandHint;
+use regex::Regex;
+use serde_json::Value;
+use std::borrow::{Borrow, Cow};
+use std::collections::HashSet;
+use std::fmt::{Display, Formatter};
+use std::hash::Hash;
+
+#[derive(Hash, Debug, PartialEq, Eq)]
+pub enum HintType {
+    FunctionType,
+    PackageType,
+    ArgumentType,
+    MethodType,
+    UnimplementedType,
+}
+
+impl Display for HintType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            FunctionType => {
+                write!(f, "Function Type")
+            }
+            PackageType => {
+                write!(f, "Package Type")
+            }
+            ArgumentType => {
+                write!(f, "Argument Type")
+            }
+            MethodType => {
+                write!(f, "Argument Type")
+            }
+            UnimplementedType => {
+                write!(f, "Unimplemented Type")
+            }
+        }
+    }
+}
+
+impl From<u64> for HintType {
+    fn from(num: u64) -> Self {
+        match num {
+            3 => FunctionType,
+            5 => ArgumentType,
+            6 => MethodType,
+            9 => PackageType,
+            _ => UnimplementedType,
+        }
+    }
+}
+
+impl Clone for HintType {
+    fn clone(&self) -> Self {
+        match self {
+            FunctionType => FunctionType,
+            PackageType => PackageType,
+            ArgumentType => ArgumentType,
+            MethodType => MethodType,
+            UnimplementedType => UnimplementedType,
+        }
+    }
+}
+
+pub fn process_completions_response(resp: &str) -> Option<HashSet<CommandHint>> {
```

Comment:
The function returns an `Option`, but there's a lot of `expect()` and `unwrap()` going on in the body. This indicates there's a lot of things that could go wrong.

Either split out the fallible stuff as a separate step (ie, parsing the json) or update the function to return a `Result` so you can bubble failure back up to the caller.